clients/admin-ui/src/features/test-datasets/DatasetEditorSection.tsx

```diff
@@ -105,28 +105,43 @@ const EditorSection = ({ connectionKey }: EditorSectionProps) => {
   };
 
   const handleSave = async () => {
+    if (!currentDataset) {
+      return;
+    }
+
+    // Parse YAML first
+    let datasetValues: Dataset;
     try {
-      if (currentDataset) {
-        const datasetValues = yaml.load(editorContent) as Partial<Dataset>;
-        const updatedDatasetConfig: DatasetConfigSchema = {
-          fides_key: currentDataset.fides_key,
-          ctl_dataset: {
-            ...currentDataset.ctl_dataset,
-            ...datasetValues,
-          },
-        };
-
-        const result = await updateDataset(updatedDatasetConfig.ctl_dataset);
-        if (isErrorResult(result)) {
-          throw new Error(getErrorMessage(result.error));
-        }
-        dispatch(setCurrentDataset(updatedDatasetConfig));
-        toast(successToastParams("Successfully modified dataset"));
-        await refetchDatasets();
-      }
-    } catch (error) {
-      toast(errorToastParams(getErrorMessage(error as FetchBaseQueryError)));
+      datasetValues = yaml.load(editorContent) as Dataset;
```

Comment:
I put this in a try/catch so that we can catch any parsing exceptions and show a toast